src/CloudFlare/IpRewrite.php

```diff
@@ -69,37 +76,28 @@ public static function reset()
     protected static function rewrite()
     {
         // only should be run once per page load
-        if (self::$is_cf !== NULL) {
+        if (self::$is_cf !== null) {
             return;
         }
-        
+
         // Set this based on the presence of the header, so it is true even if IP has already been rewritten
-        self::$is_cf = isset($_SERVER["HTTP_CF_CONNECTING_IP"]) ? TRUE : FALSE;
-        
+        self::$is_cf = isset($_SERVER['HTTP_CF_CONNECTING_IP']) ? true : false;
+
         // Store original remote address in $original_ip
-        self::$original_ip = isset($_SERVER["REMOTE_ADDR"]) ? $_SERVER["REMOTE_ADDR"] : FALSE;
+        self::$original_ip = isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : false;
 
         // Process original_ip if on cloudflare
         if (self::$is_cf && self::$original_ip) {
-            // Check for IPv4 v. IPv6
-            if (strpos(self::$original_ip, ":") === FALSE) {
-                foreach (self::$cf_ipv4 as $range) {
-                    if (IpRange::ipv4_in_range(self::$original_ip, $range)) {
-                        if (self::$is_cf) {
-                            self::$rewritten_ip = $_SERVER["REMOTE_ADDR"] = $_SERVER["HTTP_CF_CONNECTING_IP"];
-                        }
-                        break;
-                    }
-                }        
-            } else {
-                $ipv6 = IpRange::get_ipv6_full(self::$original_ip);
-                foreach (self::$cf_ipv6 as $range) {
-                    if (IpRange::ipv6_in_range($ipv6, $range)) {
-                        if (self::$is_cf) {
-                            self::$rewritten_ip = $_SERVER["REMOTE_ADDR"] = $_SERVER["HTTP_CF_CONNECTING_IP"];
-                        }
-                        break;
-                    }
+            // Check whether it's ipv4 or ipv6
+            $ip_ranges = self::$cf_ipv6;
+            if (strpos(self::$original_ip, ':') === false) {
```

Comment:
We duplicate this logic in IpUtils.  Lets make a function in IpUtils for detecting whether an IP is IPV4 or not and then use that method everywhere.